aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -2804,17 +2804,36 @@ static void linalg_lstsq_out_info(
   // now the actual call that computes the result in-place (apply_lstsq)
   at::_lstsq_helper_(solution, rank, singular_values, infos, input_working_copy, rcond, driver);
 
+  // residuals are available only if m > n and drivers other than gelsy used
   if (m > n && driver != "gelsy") {
-    // LAPACK stores residuals data for postprocessing in rows n:(m-n)
-    auto raw_residuals = solution.narrow(/*dim=*/-2, /*start=*/n, /*length*/m - n);
-    if (raw_residuals.is_complex()) {
-      // in-place abs is not supported for complex tensors
-      raw_residuals = raw_residuals.abs();
-      raw_residuals.pow_(2);
-    } else {
-      raw_residuals.abs_().pow_(2);
+    // if the driver is gelss or gelsd then the residuals are available only if rank == n
```

Comment:
This comment seems disconnected from the code around it

Reply:
I guess it would be more clear to have `if (driver == "gelss" || driver == "gelsd")` instead of `if (driver != "gels")`.